cpg_infra/billing_aggregator/driver.py

```diff
@@ -151,6 +152,45 @@ def create_source_archive(
             opts=pulumi.ResourceOptions(replace_on_changes=['*']),
         )
 
+    def setup_gcp_cost_control(self):
+        """
+        Create the gcp cost control cloud function to cut off billing when
+        it exceeds the budget
+        """
+        region = self.config.gcp.region
+        service_account = self.config.billing.gcp_cost_controls.machine_account
+        slack_channel = self.config.billing.gcp_cost_controls.slack_channel
+        pubsub_topic_name = self.config.billing.gcp_cost_controls.pubsub_topic
+
+        # Create source archive
+        source_archive = self.create_source_archive(
+            'billing-gcp-cost-report-source-code',
```

Comment:
This'll need a different resource name tot he cost report bot I think?

Reply:
Great spot. Yup changed.